nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestControlRate.java

```diff
@@ -86,18 +97,18 @@ public void testFileCountRate() throws InterruptedException {
         runner.assertQueueNotEmpty();
 
         // we have sent 3 files and after 1 second, we should be able to send the 4th
-        Thread.sleep(ONE_SEC_PLUS);
-        runner.run();
+        incrementCurrentTime();
+
+        runner.run(5);
```

Comment:
Did you find that multiple iterations were needed sometimes?

Reply:
Good question, in this case the number `5` matches the number of FlowFiles queued, which should be batched together, but this ensures that all of them should be processed.